test/e2e/serial/workload_placement_test.go

```diff
@@ -507,6 +518,135 @@ var _ = Describe("[serial][disruptive][scheduler] workload placement", func() {
+			Expect(err).ToNot(HaveOccurred())
+
+			nodes, err := e2enodes.GetWorkerNodes(fxt.Client)
+			Expect(err).ToNot(HaveOccurred())
+
+			t, _, err := taints.ParseTaints([]string{testTaint()})
+			Expect(err).ToNot(HaveOccurred())
+
+			for i := range nodes {
+				node := &nodes[i]
+				updatedNode, updated, err := taints.AddOrUpdateTaint(node, &t[0])
+				Expect(err).ToNot(HaveOccurred())
+				if updated {
+					node = updatedNode
+					klog.Infof("adding taint: %q to node: %q", t[0].String(), node.Name)
+					err = fxt.Client.Update(context.TODO(), node)
+					Expect(err).ToNot(HaveOccurred())
+				}
+			}
+		})
+
+		AfterEach(func() {
+			nodes, err := e2enodes.GetWorkerNodes(fxt.Client)
+			Expect(err).ToNot(HaveOccurred())
+
+			t, _, err := taints.ParseTaints([]string{testTaint()})
+			Expect(err).ToNot(HaveOccurred())
+
+			By("removing taints from the nodes")
+			for i := range nodes {
+				node := &nodes[i]
+				updatedNode, updated, err := taints.RemoveTaint(node, &t[0])
+				Expect(err).ToNot(HaveOccurred())
+				if updated {
+					node = updatedNode
+					klog.Infof("removing taint: %q from node: %q", t[0].String(), node.Name)
+					err = fxt.Client.Update(context.TODO(), node)
+					Expect(err).ToNot(HaveOccurred())
+				}
+			}
+
+			By("unpadding the nodes after test finish")
+			err = padder.Clean()
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		It("[test_id:47594] should make a pod with a toleration land on a node with enough resources on a specific NUMA zone", func() {
+			hostsRequired := 2
+			paddedNodes := padder.GetPaddedNodes()
+			paddedNodesSet := sets.NewString(paddedNodes...)
+
+			nrts, err := e2enrt.GetUpdated(fxt.Client, nrtv1alpha1.NodeResourceTopologyList{}, time.Second*10)
+			Expect(err).ToNot(HaveOccurred())
+
+			// so we can't support ATM zones > 2. HW with zones > 2 is rare anyway, so not to big of a deal now.
+			By(fmt.Sprintf("filtering available nodes with at least %d NUMA zones", 2))
+			nrtCandidates := e2enrt.FilterZoneCountEqual(nrts.Items, 2)
+			if len(nrtCandidates) < hostsRequired {
+				Skip(fmt.Sprintf("not enough nodes with 2 NUMA Zones: found %d", len(nrtCandidates)))
+			}
+
+			singleNUMAPolicyNrts := e2enrt.FilterSingleNumaNodePolicyNrts(nrts.Items)
+			nodesNameSet := e2enrt.AccumulateNames(singleNUMAPolicyNrts)
+
+			// the only node which was not padded is the targetedNode
+			// since we know exactly how the test setup looks like we expect only targeted node here
+			targetNodeNameSet := nodesNameSet.Difference(paddedNodesSet)
+			Expect(targetNodeNameSet.Len()).To(Equal(1))
+
+			targetNodeName, ok := targetNodeNameSet.PopAny()
+			Expect(ok).To(BeTrue())
+
+			testPod := objects.NewTestPodPause(fxt.Namespace.Name, "testpod")
+			pSpec := &testPod.Spec
+			if pSpec.Tolerations == nil {
+				pSpec.Tolerations = testToleration()
+			} else {
+				pSpec.Tolerations = append(pSpec.Tolerations, testToleration()...)
+			}
```

Comment:
more idiomatic:
1. create empty `tolerations` if `nil`
2. always add the tolerations we want (now it's safe to do because guaranteed non-nil)

Reply:
nice!